### PR TITLE
Upgraded k8s dependencies version to 0.26.1

### PR DESCRIPTION
**Pull request by shahpratikr**
## Change Overview

This PR upgrades
- K8s dependency version to 0.26.1
- Kopia to latest commit 5c901abc90857d4b85e6694ef4eeb86de7e5e89b

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

GitHub Actions were successful.

### CODE REVIEW

**Comment by r4rajat** [comment]
LGTM

**Comment by kale-amruta** [comment]
@shahpratikr please test it against K10 to see if updating k8s doesnt break k10

In this PR - https://github.com/kanisterio/kanister/pull/1619 , we have added a comment `ok-to-test` so that it runs a pipeline to test the changes with K10. @akankshakumari393 can you please let Pratik know which pipeline gets triggered?

File: go.mod [hunk 1]
```diff
@@ -32,7 +31,7 @@ require (
 	github.com/hashicorp/go-version v1.6.0
 	github.com/jpillora/backoff v1.0.0
 	github.com/json-iterator/go v1.1.12
-	github.com/kopia/kopia v0.10.7-0.20220330221649-08cf7eb936c6
+	github.com/kopia/kopia v0.12.2-0.20221217161912-f8be8f6a5698
```
Comment (viveksinghggits) [inline go.mod:34]: let's not include kopia upgrade in this PR. Can we raise another PR just for Kopia upgrade and in this PR we can just have the k8s deps upgrade?

Comment (shahpratikr) [reply go.mod:34]: K8s dependency upgrades some otel packages which throws error while running `go mod tidy`. These packages are dependent on each other. Hence we need to upgrade both in same PR

Comment (viveksinghggits) [reply go.mod:34]: @ankitjain235 what do you think about this chaneg? PR looks good otherwise.

Comment (ankitjain235) [reply go.mod:34]: Yes, this would be required.
We are just switching from https://github.com/kastenhq/kopia/commit/f1fc864f1dd09aa5103ead47f655ee302a5117ce which is some temporary commits above https://github.com/kopia/kopia/commit/f8be8f6a5698edba055291b138c117ec6eaae125 see https://github.com/kastenhq/kopia/commits/get-repo-upgrade-validate.
We are switching to https://github.com/kopia/kopia/commit/f8be8f6a5698edba055291b138c117ec6eaae125 which is effectively same.

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.19
 replace (
 	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
 	github.com/graymeta/stow => github.com/kastenhq/stow v0.2.6-kasten.1.0.20220726203146-8a90401257d4
-	github.com/kopia/kopia => github.com/kastenhq/kopia v0.0.0-20221221212937-f1fc864f1dd0
+	github.com/kopia/kopia => github.com/kastenhq/kopia v0.0.0-20230209084412-6e05ca94108a
```
Comment (pavannd1) [inline go.mod:8]: Are we switching to upstream kopia in a different PR?

File: go.mod (hunk 1 above)
Comment (pavannd1) [reply go.mod:35]: IIUC The kopia dependency is getting replaced by `kastenhq/kopia` in the `replace` directive above. Not sure why this change is required

File: go.mod (hunk 2 above)
Comment (shahpratikr) [reply go.mod:8]: Upstream kopia doesn't have a fix for some files which were included in forked repo. Until we raise a PR or they fix it, we cannot change to upstream kopia

File: build/golint.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@ TIMEOUT="10m"
 
 echo "Running golangci-lint..."
 
-golangci-lint run --timeout ${TIMEOUT} --skip-dirs ${SKIP_DIR_REGEX} -E maligned,whitespace,gocognit,unparam -e '`ctx` is unused'
+golangci-lint run --timeout ${TIMEOUT} --skip-dirs ${SKIP_DIR_REGEX} -E govet,whitespace,gocognit,unparam -e '`ctx` is unused'
```
Comment (PrasadG193) [inline build/golint.sh:25]: Why is this change required?

File: pkg/kube/exec.go [hunk 4]
```diff
@@ -147,7 +148,7 @@ func execute(method string, url *url.URL, config *restclient.Config, stdin io.Re
 	if err != nil {
 		return err
 	}
-	return exec.Stream(remotecommand.StreamOptions{
+	return exec.StreamWithContext(context.TODO(), remotecommand.StreamOptions{
```
Comment (PrasadG193) [inline pkg/kube/exec.go:151]: We should propagate context from parent functions and pass it here.

File: build/golint.sh (hunk 3 above)
Comment (shahpratikr) [reply build/golint.sh:25]: maligned is deprecated in current version of golangci-lint we use.

File: pkg/kube/exec.go (hunk 4 above)
Comment (shahpratikr) [reply pkg/kube/exec.go:151]: we are not getting any context from parent function. if we decide to provide one, it might break things. hence created a new context here

File: go.mod (hunk 1 above)
Comment (shahpratikr) [reply go.mod:35]: as discussed, even though we are not using upstream kopia version, this line updates kopia version and commit. this helps developers to figure out the details of upstream kopia we are using. hence not reverting the change

File: pkg/kube/exec.go (hunk 4 above)
Comment (pavannd1) [reply pkg/kube/exec.go:151]: We can do it in a follow up 👍🏼

File: build/golint.sh (hunk 3 above)
Comment (PrasadG193) [reply build/golint.sh:25]: Is `govet` a replacement of this?

File: pkg/kube/exec.go (hunk 4 above)
Comment (PrasadG193) [reply pkg/kube/exec.go:151]: @shahpratikr Please file a JIRA for that.

File: build/golint.sh (hunk 3 above)
Comment (pavannd1) [reply build/golint.sh:25]: It's good to provide references or links for such changes
```
maligned ⚙️ ⚠ | The repository of the linter has been archived by the owner. Replaced by govet 'fieldalignment'.
```
https://github.com/mdempsky/maligned

Comment (shahpratikr) [reply build/golint.sh:27]: added comment

File: pkg/kube/exec.go (hunk 4 above)
Comment (shahpratikr) [reply pkg/kube/exec.go:152]: Added link in a comment

File: pkg/kube/exec.go [hunk 5]
```diff
@@ -147,7 +148,8 @@ func execute(method string, url *url.URL, config *restclient.Config, stdin io.Re
 	if err != nil {
 		return err
 	}
-	return exec.Stream(remotecommand.StreamOptions{
+	// Get context from a caller function. Ticket to track: https://kasten.atlassian.net/browse/K10-15138
```
Comment (pavannd1) [inline pkg/kube/exec.go:151]: Please remove this link. Create a github issue and link that instead

Comment (shahpratikr) [reply pkg/kube/exec.go]: fixed

File: build/golint.sh [hunk 6]
```diff
@@ -22,7 +22,9 @@ TIMEOUT="10m"
 
 echo "Running golangci-lint..."
 
-golangci-lint run --timeout ${TIMEOUT} --skip-dirs ${SKIP_DIR_REGEX} -E maligned,whitespace,gocognit,unparam -e '`ctx` is unused'
+# https://github.com/mdempsky/maligned is deprecated.
+# maligned ⚙️ ⚠ | The repository of the linter has been archived by the owner. Replaced by govet 'fieldalignment'.
```
Comment (viveksinghggits) [inline build/golint.sh:26]:
```suggestion
# maligned | The repository of the linter has been archived by the owner. Replaced by govet 'fieldalignment'.
```

**Review by viveksinghggits — COMMENTED** [review]
LGTM

File: build/golint.sh (hunk 6 above)
Comment (shahpratikr) [reply build/golint.sh]: fixed

**Review by pavannd1 — APPROVED** [review]

**Review by viveksinghggits — APPROVED** [review]
